apps/examples/testcase/le_tc/compression/tc_compress_read.c

```diff
@@ -22,12 +22,51 @@
 #include <tinyara/config.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <errno.h>
 #include <fcntl.h>
 #include <sys/ioctl.h>
 #include <tc_common.h>
 #include <tinyara/os_api_test_drv.h>
+#include <tinyara/compression.h>
 #include "tc_internal.h"
 
+#define MAX_INPUT_SIZE 10000
+#define MAX_OUTPUT_SIZE 10000
+#define ERROR -1
+
+static struct compress_header *comp_tc_data;
+static struct compress_header *decomp_tc_data;
+
+void uninitialize_tc_data()
+{
+	free(comp_tc_data->input_buffer);
+	free(comp_tc_data->output_buffer);
+	free(decomp_tc_data->input_buffer);
+	free(decomp_tc_data->output_buffer);
+	free(comp_tc_data);
+	free(decomp_tc_data);
+}
+
+void initialize_tc_data()
+{
+	comp_tc_data = (struct compress_header *)malloc(sizeof(struct compress_header));
+	comp_tc_data->input_buffer = (unsigned char *)malloc(MAX_INPUT_SIZE);
+	comp_tc_data->input_size = MAX_INPUT_SIZE;
+	for( int i = 0 ; i < comp_tc_data->input_size; i++){
+		comp_tc_data->input_buffer[i] = 'a' + (rand() % 26) ;
+	}
+	comp_tc_data->output_size = MAX_OUTPUT_SIZE;
+	comp_tc_data->output_buffer = (unsigned char *)malloc(MAX_OUTPUT_SIZE);
+	
+	decomp_tc_data = (struct compress_header *)malloc(sizeof(struct compress_header));
+	decomp_tc_data->input_buffer = comp_tc_data->output_buffer;
+	decomp_tc_data->input_size = comp_tc_data->output_size;
+	decomp_tc_data->output_size = comp_tc_data->input_size;
+	decomp_tc_data->output_buffer = (unsigned char *)malloc(comp_tc_data->input_size);
+	
+	return ;		
```

Comment:
unnecessary. This will be a SVACE defect.

Reply:
done